src/emerge.cpp

```diff
@@ -722,6 +725,11 @@ void *EmergeThread::run()
 		if (block)
 			modified_blocks[pos] = block;
 
+		if (block && (bedata.flags & BLOCK_EMERGE_ACTIVATE)) {
+			MutexAutoLock envlock(m_server->m_env_mutex);
+			m_server->m_env->activateObjects(block, 0);
```

Comment:
Note that this is only activating the objects.
Could activate the block too, but (a) that seemed unnecessary, and (b) the block will be activated again when the serverenvironment detects it as a new block.